drivers/hmis/app/models/hmis/form/processor.rb

```diff
@@ -0,0 +1,196 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class Hmis::Form::Processor < ::GrdaWarehouseBase
+  self.table_name = :hmis_assessment_processors
+
+  has_one :assessment_detail
+
+  # assessment is accessed through the assessment_detail
+  belongs_to :health_and_dv, class_name: 'Hmis::Hud::HealthAndDv', primary_key: :HealthAndDVID, optional: true
+  belongs_to :income_benefit, class_name: 'Hmis::Hud::IncomeBenefit', primary_key: :IncomeBenefitID, optional: true
+  belongs_to :enrollment_coc, class_name: 'Hmis::Hud::EnrollmentCoc', primary_key: :EnrollmentCocID, optional: true
+  belongs_to :physical_disability, class_name: 'Hmis::Hud::Disability', primary_key: :DisabilityID, optional: true
+  belongs_to :developmental_disability, class_name: 'Hmis::Hud::Disability', primary_key: :DisabilitiesID, optional: true
+  belongs_to :chronic_health_condition, class_name: 'Hmis::Hud::Disability', primary_key: :DisabilitiesID, optional: true
+  belongs_to :hiv_aids, class_name: 'Hmis::Hud::Disability', primary_key: :DisabilitiesID, optional: true
+  belongs_to :mental_health_disorder, class_name: 'Hmis::Hud::Disability', primary_key: :DisabilitiesID, optional: true
+  belongs_to :substance_use_disorder, class_name: 'Hmis::Hud::Disability', primary_key: :DisabiliesID, optional: true
+
+  def run!
+    return unless assessment_detail.hud_values.present?
+
+    assessment_detail.hud_values.each do |key, value|
+      # Don't use greedy matching so that the container is up to the first dot, and the rest is the field
+      container, field = /(.*?)\.(.*)/.match(key)[1..2]
+
+      container_processor(container)&.process(field, value)
+    end
+  end
+
+  # Type Factories
+  def enrollment_factory
+    # The enrollment has already been created, so we can just return it
+    assessment_detail.assessment.enrollment
+  end
+
+  # The items associated with the enrollments are all singletons, so return
+  # them if they already exist, otherwise create them
+  def enrollment_coc_factory
+    return enrollment_coc if enrollment_coc.present?
+
+    self.enrollment_coc = enrollment_factory.enrollment_cocs.
+      where(data_collection_stage: assessment_detail.data_collection_stage).
+      first_or_initialize(
+        enrollment_id: enrollment_factory.enrollment_id,
```

Comment:
Pretty sure all of these will need `data_source_id`

Reply:
Turns out the `enrollment_id` and `data_source_id` get set via Rails magic (they are declared as the keys in the `has_many`) -- so I went the other way, and took `enrollment_id` out.